zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/UserTaskQueryController.java

```diff
@@ -60,4 +68,29 @@ private ResponseEntity<Object> search(final UserTaskQuery query) {
       return RestErrorMapper.mapProblemToResponse(problemDetail);
     }
   }
+
+  @GetMapping(
+      path = "/{userTaskKey}",
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE})
+  public ResponseEntity<UserTaskItem> getByKey(
+      @PathVariable("userTaskKey") final Long userTaskKey) {
+    try {
+      return ResponseEntity.ok()
+          .contentType(new MediaType(MediaType.APPLICATION_JSON, StandardCharsets.UTF_8))
+          .body(SearchQueryResponseMapper.toUserTask(userTaskServices.getByKey(userTaskKey)));
+    } catch (final NotFoundException nfe) {
+      final var problemDetail =
+          RestErrorMapper.createProblemDetail(
+              HttpStatus.NOT_FOUND, nfe.getMessage(), RejectionType.NOT_FOUND.name());
+      return RestErrorMapper.mapProblemToResponse(problemDetail);
+    } catch (final Exception e) {
+      REST_LOGGER.warn("An exception occurred in Get User Task by key.", e);
+      final var problemDetail =
+          RestErrorMapper.createProblemDetail(
+              HttpStatus.INTERNAL_SERVER_ERROR,
+              e.getMessage(),
+              "Failed to execute Get User Task by key.");
+      return RestErrorMapper.mapProblemToResponse(problemDetail);
+    }
```

Comment:
🔧 My recommendation here would be to use `RestErrorMapper::mapErrorToProblem` for any exception that occurs when calling the `UserTaskServices::getByKey`. We just need to add the `NotFoundException` to the switch statement in the `RestErrorMapper`. The generic exception case will change a little bit regarding messaging when using the mapper instead (it runs into the default case there then). We can either live with that or adjust the `RestErrorMapper::mapErrorToProblem` by adding a String parameter that allows defining the message in the generic error case.

Reply:
Great catch 💯  @tmetzke  :) 
I just updated the Controller with your suggested change

I am going to apply the same update on Form ASAP as I merge this PR.

Reply:
@tmetzke  I have applied the follow suggestions. Please check if the `NotFoundException` is according to the guidelines

Following the implementation we will have the follow error for Not found

```
                {
                  "type": "about:blank",
                  "status": 404,
                  "title": "NOT_FOUND",
                  "detail": "User Task with key 999 not found"
                }
```

Reply:
👍 Very nice